app/src/main/java/ua/in/badparking/ui/fragments/LocationFragment.java

```diff
@@ -91,9 +90,10 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         Location location = lm.getLastKnownLocation(LocationManager.NETWORK_PROVIDER);
         geocoder = new Geocoder(getActivity(), Locale.getDefault());
 
-        if (location != null) {
+        if (location != null && currentLocation == null) {
```

Comment:
We update the marker in onMapReady and in receiver
if we haven't  currentLocation - marker will not update (until we get the first position in the receiver)
but it is a very rare situation (for example, user reinstall android or fresh buying)
in this case, the user will receive a message that he need to wait a little longer to get the first GPS coordinates and marker will be show and update